### ISSUE TITLE
Make components work when rendered inside of iframes

### ISSUE THREAD

**Issue by giuseppeg** [reach/reach-ui#112]
Some components reference the global `document`/`window` object directly. This means that when a page renders a React app inside of an iframe (eg. widgets or entire applications) Reach UI references the parent frame's objects instead. 

The most obvious example is `WindowSize`, when used inside of an iframe it returns the parent window size instead of the `ownerDocument`'s. See https://codesandbox.io/s/32q7nwojp5

I believe that other components might come with similar issues. Eg. the positioning logic of the menu or the tab trap logic.

I would love to evaluate and integrate Reach at work if it turns out to provide better solutions/accessibility than ours implementations, but without iframes-compatibility I cannot even try the components. I'd be glad to help. 

(Note: React always takes this scenario into account)

**Comment by giuseppeg** [comment]
Example #137

**Comment by chancestrickland** [comment]
I know we addressed this in @reach/portal, but it'd probably be good for @reach/window-size to allow the user to decide whether to use a global `window` vs. the `ownerDocument.defaultView`.

**Comment by Kikketer** [comment]
Having this same issue using react-frame and the @reach/menu-button.  The portal it creates lands on the document parent instead of the frame.  I've seen this issue with a couple libraries I've tried, most don't account for this type of scenario (probably with good reason since it's probably rare).

I noticed that the menu-button simply uses `<Portal>`.  Wrapping my head around how the `component-component` works, would this situation be as easy as adding the `mountNode` to the params of `menu-button`?  Then passing that into the `<Portal mountNode={mountNode}>`?

**Comment by Kikketer** [comment]
So I snagged (I think version 0.5.0) of the menu-button.js and adapted it to work inside an iFrame by passing in the `document` property.  I'm a bit overloaded at work right now so I can't make a proper PR but I did copy/paste it into a gist:

https://gist.github.com/Kikketer/b22d5eb9398a30c58cc71b817e018085#file-menu-button-js-L95

This works all the way down to IE11.  Passing in the document and then using that if it's set allows the "click anywhere to close" the menu to work properly when in an iFrame.  The only thing left is having it adjust properly based on size of that iFrame.

Hopefully this helps a little bit and maybe inspires for `Hacktoberfest`?
